torchtext/experimental/datasets/raw/text_classification.py

```diff
@@ -0,0 +1,250 @@
+import torch
+import io
+from torchtext.utils import download_from_url, extract_archive, unicode_csv_reader
+
+URLS = {
+    'AG_NEWS':
+        'https://drive.google.com/uc?export=download&id=0Bz8a_Dbh9QhbUDNpeUdjb0wxRms',
+    'SogouNews':
+        'https://drive.google.com/uc?export=download&id=0Bz8a_Dbh9QhbUkVqNEszd0pHaFE',
+    'DBpedia':
+        'https://drive.google.com/uc?export=download&id=0Bz8a_Dbh9QhbQ2Vic1kxMmZZQ1k',
+    'YelpReviewPolarity':
+        'https://drive.google.com/uc?export=download&id=0Bz8a_Dbh9QhbNUpYQ2N3SGlFaDg',
+    'YelpReviewFull':
+        'https://drive.google.com/uc?export=download&id=0Bz8a_Dbh9QhbZlU4dXhHTFhZQU0',
+    'YahooAnswers':
+        'https://drive.google.com/uc?export=download&id=0Bz8a_Dbh9Qhbd2JNdDBsQUdocVU',
+    'AmazonReviewPolarity':
+        'https://drive.google.com/uc?export=download&id=0Bz8a_Dbh9QhbaW12WVVZS2drcnM',
+    'AmazonReviewFull':
+        'https://drive.google.com/uc?export=download&id=0Bz8a_Dbh9QhbZVhsUnRWRDhETzA',
+    'IMDB':
+        'http://ai.stanford.edu/~amaas/data/sentiment/aclImdb_v1.tar.gz'
+}
+
+
+def _create_data_from_csv(data_path):
+    data = []
+    with io.open(data_path, encoding="utf8") as f:
+        reader = unicode_csv_reader(f)
+        for row in reader:
+            data.append((int(row[0]), ' '.join(row[1:])))
+    return data
+
+
+class RawTextDataset(torch.utils.data.Dataset):
```

Comment:
In light of multiprocessing, data parallelism etc. etc. I wonder if it's worthwhile looking into making this an IterableDataset after all and introducing a convenience function that creates a map-style dataset by simply exhausting the iterator and writing out the result. After all this is the raw dataset, so that'd provide an even more general interface. Let's also look into JIT-ability of that.

Reply:
Switch to `IterableDataset` for raw text dataset.